src/lsdb/nested/datasets/generation.py

```diff
@@ -53,22 +182,39 @@ def generate_parquet_file(n_base, n_layer, path, file_per_layer=True, npartition
         The number of rows per n_base row to generate for a nested layer.
         Alternatively, a dictionary of layer label, layer_size pairs may be
         specified to created multiple nested columns with custom sizing.
-    path : str,
-        The path to the parquet file to write to if `file_per_layer` is `False`,
-        and otherwise the path to the directory to write the parquet file for
-        each layer.
-    file_per_layer : bool, default=True
-        TODO: Currently only True is supported.
-        If True, write each layer to its own parquet file. Otherwise, write
-        the generated to a single parquet file representing a nested dataset.
-    npartitions : int, default=1
-        The number of Dask partitions to split the generated data into for each layer.
-    seed : int, default=None
+    seed : int
         A seed to use for random generation of data
+    ra_range : tuple
+        A tuple of the min and max values for the ra column in degrees
+    dec_range : tuple
+        A tuple of the min and max values for the dec column in degrees
+    search_filter : AbstractSearch
+        A search filter to apply to the generated data. Currently supports the
+        ConeSearch and BoxSearch filters. Note that if provided,
+        this will override the `ra_range` and `dec_range` parameters.
+    **kwargs :
+        Additional keyword arguments to pass to `lsdb.from_dataframe`.
 
     Returns
     -------
-    None
+    Catalog
+        The constructed LSDB Catalog.
+
+    Examples
+    --------
+    >>> from lsdb.nested.datasets import generate_catalog # doctest: +SKIP
+    >>> generate_catalog(10,100) # doctest: +SKIP
+    >>> generate_catalog(1000, 10, ra_range=(0.,10.), dec_range=(-5.,0.)) # doctest: +SKIP
+
+    Constraining spatial ranges:
+    >>> generate_data(10, 100, ra_range=(0., 10.), dec_range=(-5., 0.)) # doctest: +SKIP
+
+    Using a search filter:
+    >>> from lsdb.core.search import ConeSearch # doctest: +SKIP
+    >>> generate_data(10, 100, search_filter=ConeSearch(5, 5, 1)) # doctest: +SKIP
     """
-    nf = generate_data(n_base, n_layer, npartitions, seed)
-    nf.to_parquet(path, by_layer=file_per_layer, write_index=False)
+
+    base_nf = generate_data(
+        n_base, n_layer, seed=seed, ra_range=ra_range, dec_range=dec_range, search_filter=search_filter
+    )
+    return lsdb.from_dataframe(base_nf.compute(), catalog_name="generated_catalog", **kwargs)
```

Comment:
If I pass `catalog_name="fun_times"` as a kwarg, which one wins?

Reply:
Good catch, updated this to just name it "generated_catalog" if a name isn't provided, but no longer blocks the use of a custom name